trio/_highlevel_open_tcp_listeners.py

```diff
@@ -1,11 +1,18 @@
+from __future__ import annotations
+
 import errno
 import sys
+from collections.abc import Awaitable, Callable
 from math import inf
+from typing import TYPE_CHECKING
 
 import trio
 
 from . import socket as tsocket
 
+if TYPE_CHECKING:
+    from trio.lowlevel import TaskStatus
```

Comment:
Does this import need to be guarded (due to import cycles), or can it be a real import? It's probably better to just import if we can, so that this can be evaluated at runtime.

Reply:
Does not need to be guarded

Reply:
https://github.com/python-trio/trio/issues/2687

This is kinda up in the air, I'd love it if we definitively resolved it and added a CI check that made usage consistent and saved review cycles. https://pypi.org/project/flake8-type-checking/

Reply:
I would suggest using https://pypi.org/project/ruff/ instead, it does the same thing and more but faster and with auto-fix support